source/jormungandr/jormungandr/georef.py

```diff
@@ -90,15 +86,21 @@ def get_physical_mode(self, uri, request_id):
             return res.physical_modes[0]
         return None
 
-    def get_car_co2_emission(self, distance, request_id):
+    def get_car_co2_emission(self, request_id):
         logger = logging.getLogger(__name__)
 
-        car_mode = self.get_physical_mode('physical_mode:Car', request_id)
-        if car_mode is None or not car_mode.HasField('co2_emission_rate'):
-            logger.warning("Cannot compute car co2 emission with the distance {}".format(distance))
-            return response_pb2.Co2Emission(value=0, unit='gEC')
-
-        return response_pb2.Co2Emission(value=car_mode.co2_emission_rate.value * distance / 1000.0, unit='gEC')
+        @memory_cache.memoize(
+            current_app.config[str('MEMORY_CACHE_CONFIGURATION')].get(str('TIMEOUT_AUTHENTICATION'), 30)
+        )
+        @cache.memoize(current_app.config[str('CACHE_CONFIGURATION')].get(str('TIMEOUT_AUTHENTICATION'), 300))
+        def inner(mode_str):
```

Comment:
Instead of using this convoluted solution of an inner function, to ignore the `request_id` from the cache key, `@cache.memoize` has an option to ignore some function parameters with `args_to_ignore` (see [documentation](https://flask-caching.readthedocs.io/en/latest/api.html#flask_caching.Cache.memoize)).

So the PR could only be

```
        @memory_cache.memoize(
            current_app.config[str('MEMORY_CACHE_CONFIGURATION')].get(str('TIMEOUT_AUTHENTICATION'), 30),
            args_to_ignore=["request_id"],
        )
        @cache.memoize(
            current_app.config[str('CACHE_CONFIGURATION')].get(str('TIMEOUT_AUTHENTICATION'), 300), 
            args_to_ignore=["request_id"],
        )
```